### PR TITLE
Document Chronograf basic auth

### LINKED ISSUE #1557: Document Chronograf basic auth options
https://github.com/influxdata/chronograf/pull/5577

### PR DESCRIPTION
**Pull request by pierwill**
Closes #1557.

Also updates some older formatting for notes.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/chronograf/v1.8/administration/managing-security.md [hunk 1]
```diff
@@ -33,21 +35,23 @@ To configure any of the supported OAuth 2.0 providers to work with Chronograf,
...
+1. Generate a high-entropy pseudo-random string.
 
-    > For example, to do this with OpenSSL, run this command:
-    > ```
-    > openssl rand -base64 256 | tr -d '\n'
-    > ```
+   For example, to do this with OpenSSL, run this command:
+   ```sh
+   openssl rand -base64 256 | tr -d '\n'
+   ```
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:43]: indent lines 40-43
(line 45 currently renders as number 1)

File: content/chronograf/v1.8/administration/managing-security.md [hunk 2]
```diff
@@ -76,7 +80,7 @@ Configuration steps for the following supported authentication providers are pro
 * [Okta](#configure-okta-authentication)
 * [Gitlab](#configure-gitlab-authentication)
 * [Azure Active Directory](#configure-azure-active-directory-authentication)
-* [Configure Chronograf to use any OAuth 2.0 provider](#configure-chronograf-to-use-any-oauth-2-0-provider)
+* [Configure Chronograf to use any OAuth 2.0 provider](#configure-chronograf-to-use-any-oauth-20-provider)
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:83]: ...oauth-2-0-provider)

File: content/chronograf/v1.8/administration/managing-security.md [hunk 3]
```diff
@@ -417,7 +421,9 @@ JWKS_URL="https://example.com/adfs/discovery/keys"
 TOKEN_SECRET="ZNh2N9toMwUVQxTVEe2ZnnMtgkh3xqKZ"
 ```
 
-> _**Note:**_ Do not use special characters for the GENERIC_CLIENT_ID as AD FS will split strings here, finally resulting in an identifier mismatch.
+{{% note %}}
+Do not use special characters for the `GENERIC_CLIENT_ID` as AD FS will split strings here, finally resulting in an identifier mismatch.
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:425]: Shall we spell out Active Directory Federation Services?

will split > splits

splits a string at the special character?

does omitting "finally" change the meaning?

File: content/chronograf/v1.8/administration/managing-security.md [hunk 4]
```diff
@@ -441,6 +447,24 @@ export AUTH_DURATION=1080h
 To require re-authentication every time the browser is closed, set `AUTH_DURATION` to `0`.
 This makes the cookie transient (aka "in-memory").
 
+## Configure Chronograf to authenticate with a username and password
+
+Chronograf can be configured to authenticate users by username and password ("basic authentication").
+
+{{% warn %}}
+[OAuth 2.0](#configure-chronograf-to-authenticate-with-oauth-20) is the prefered method for authentication.
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:455]: preferred

File: content/chronograf/v1.8/administration/managing-security.md [hunk 5]
```diff
@@ -441,6 +447,24 @@ export AUTH_DURATION=1080h
...
+Only use basic authentication in cases where an OAuth 2.0 integration is not possible.
+{{% /warn %}}
+
+When using basic authentication, *all users have SuperAdmin status*.
+With basic authentication enabled, Chronograf authorization rules are not enforced.
+For more information, see [Cross-organization SuperAdmin status](/chronograf/v1.8/administration/managing-chronograf-users/#cross-organization-superadmin-status).
+
+Turn on HTTP basic authentication access to restrict HTTP requests to Chronograf to selected users.
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:463]: Is basic authentication always "HTTP" basic authentication in this context? Wondering if we should clarify this elsewhere, or simply say "Turn on basic auth..." here?

Comment (kelseiv) [reply content/chronograf/v1.8/administration/managing-security.md:463]: To turn on....to selected users, open the `chronograf` CLI, and run `chronograf --htpasswd.....` (to be sure, is there something missing between chronograf and --htpasswd?)

should we clarify how to set up `HTPASSWD`?

File: content/chronograf/v1.8/administration/managing-security.md [hunk 6]
```diff
@@ -441,6 +447,24 @@ export AUTH_DURATION=1080h
...
+Use the Chronograf CLI with `—htpasswd <path to .htpasswd file>` or `HTPASSWD` environment variable.
+The `.htpasswd` file contains users and their passwords.
+For more information about managing the htpasswd utility, see [NGINX documentation](https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/).
+
```
Comment (kelseiv) [inline content/chronograf/v1.8/administration/managing-security.md:467]: For more information about how to restrict access with HTTP basic authentication, see [NG....

File: content/chronograf/v1.8/tools/chronograf-cli/_index.md [hunk 7]
```diff
@@ -68,11 +68,12 @@ chronograf [flags]
...
+| `--auth-duration`                    | Total duration, in hours, of cookie life for authentication. Default value is `720h`.                                                                                                                                                                                                               | `$AUTH_DURATION` |
+| `--public-url`                       | Public URL required to access Chronograf using a web browser. For example, if you access Chronograf using the default URL, the public URL value would be `http://localhost:8888`. Required for Google OAuth 2.0 authentication. Used for Auth0 and some generic OAuth 2.0 authentication providers. | `$PUBLIC_URL`    |
+| `—htpasswd <path to .htpasswd file>` | Password file for use with HTTP basic authentication (username and password).                                                                                                                                                                                                                       | `$HTPASSWD`      |
```
Comment (kelseiv) [inline content/chronograf/v1.8/tools/chronograf-cli/_index.md:76]: Let's link to nginx doc here too & maybe describe what the htpasswd file is/whether it's installed with....nginx/Chronograf?

**Review by kelseiv — CHANGES_REQUESTED** [review]

File: content/chronograf/v1.8/administration/managing-security.md [hunk 8]
```diff
@@ -10,14 +10,16 @@ menu:
 ---
 
 To enhance security, configure Chronograf to authenticate and authorize with [OAuth 2.0](https://oauth.net/) and use TLS/HTTPS.
+(Basic authentication is also available.)
 
-* [Configure OAuth 2.0](#configure-oauth-2-0)
+* [Configure Chronograf to authenticate with OAuth 2.0](#configure-chronograf-to-authenticate-with-oauth-20)
```
[the remark this answers is not in the document]
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:15]: With this internal link,`oauth-2-0` is not working for me locally. Only `oauth-20` works... Perhaps something has changed in Hugo?

File: content/chronograf/v1.8/administration/managing-security.md (hunk 3 above)
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:425]: We define this above in the same document.

File: content/chronograf/v1.8/administration/managing-security.md (hunk 5 above)
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:463]: It looks fairly involved and requires research into more third-party tools, it seems. Let's link out for now and come back if folks ask for more?

File: content/chronograf/v1.8/administration/managing-security.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (kelseiv) [reply content/chronograf/v1.8/administration/managing-security.md:15]: _gee, I hope not too!_ What version are you on?

When I run locally, I see this:
image

I'm on Hugo v0.59.1

@sanderson, will you run this branch locally and let us know what you see?

File: content/chronograf/v1.8/administration/managing-security.md (hunk 1 above)
Comment (kelseiv) [reply content/chronograf/v1.8/administration/managing-security.md:43]: @pierwill, when I run locally, I still see line 45 as #1 

image

File: content/chronograf/v1.8/administration/managing-security.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sanderson) [reply content/chronograf/v1.8/administration/managing-security.md:15]: @pierwill upgrade to 0.75.0 to match what's being used in the deploy pipeline. I upgraded everything to be compatible with 0.75.0 before I left. Hugo may have changed the way it generates anchors.

File: content/chronograf/v1.8/administration/managing-security.md (hunk 1 above)
Comment (sanderson) [reply content/chronograf/v1.8/administration/managing-security.md:43]: To ensure content is properly nested in a list item (and not break the list numbering), indent the nested content with 4 space, not 2.

File: content/chronograf/v1.8/administration/managing-security.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sanderson) [reply content/chronograf/v1.8/administration/managing-security.md:15]: It should be `...-2-0`

[the remark this answers is not in the document]
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:15]: Huh, I'm using `Hugo Static Site Generator v0.76.5/extended darwin/amd64 BuildDate: unknown` from Homebrew. The anchors do not work for me when using `...-2-0`.

[the remark this answers is not in the document]
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:15]: Reverting to 0.75 now...

[the remark this answers is not in the document]
Comment (pierwill) [reply content/chronograf/v1.8/administration/managing-security.md:15]: It only works *without the dash* for me in both 0.75 and 0.76.5.

**Pull request merged**